metrics/cmd/heapster-apiserver/app/heapstermetrics.go

```diff
@@ -21,19 +21,45 @@ import (
 	metricsink "github.com/kubernetes-incubator/metrics-server/metrics/sinks/metric"
 	nodemetricsstorage "github.com/kubernetes-incubator/metrics-server/metrics/storage/nodemetrics"
 	podmetricsstorage "github.com/kubernetes-incubator/metrics-server/metrics/storage/podmetrics"
+	"k8s.io/apimachinery/pkg/apimachinery/announced"
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	v1listers "k8s.io/client-go/listers/core/v1"
-	"k8s.io/client-go/pkg/api"
 	"k8s.io/metrics/pkg/apis/metrics"
-	_ "k8s.io/metrics/pkg/apis/metrics/install"
+	"k8s.io/metrics/pkg/apis/metrics/install"
 	"k8s.io/metrics/pkg/apis/metrics/v1alpha1"
 )
 
+var (
+	groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+	registry             = registered.NewOrDie("")
+	Scheme               = runtime.NewScheme()
+	Codecs               = serializer.NewCodecFactory(Scheme)
+)
+
 func installMetricsAPIs(s *options.HeapsterRunOptions, g *genericapiserver.GenericAPIServer,
 	metricSink *metricsink.MetricSink, nodeLister v1listers.NodeLister, podLister v1listers.PodLister) {
+	install.Install(groupFactoryRegistry, registry, Scheme)
```

Comment:
As I recall, that was satisfied by putting the init block in this file to ensure the Scheme was always complete.  Having one scheme per API server seems very reasonable to me.  If we want to eliminate it entirely, then move these one layer out and construct them inside of the command which may be cleaner in the long run.

We're still feeling through these aspects, but we know for sure that we want separate schemes to avoid unintended side-effects as people import multiple projects.